APMrover2/mode.cpp

```diff
@@ -216,6 +216,16 @@ void Mode::set_desired_speed_to_default(bool rtl)
     _desired_speed = get_speed_default(rtl);
 }
 
+// set desired speed in m/s
+bool Mode::set_desired_speed(float speed)
+{
+    if (!is_negative(speed)) {
```

Comment:
This doesn't seem right to me.  It should be possible to indicate you want to drive backwards using a negative speed.

Reply:
@peterbarker, thanks for the review.  I agree and I actually tested it but it doesn't work at the moment.  Instead of turning around and reversing towards the waypoint, it instead backs away from the waypoint forever.  I think perhaps we should start with the simple solution and handle the reversing part in a follow up PR at some point.

Reply:
Sounds good.  Perhaps an issue - sounds like a good-first-issue thing...